lib/AssemblyUtil/FastaToAssembly.py

```diff
@@ -42,10 +74,16 @@ def import_fasta(self, params):
         mass_params = self._set_up_single_params(params)
         return self._import_fasta_mass(mass_params)[0]
 
-    def import_fasta_mass(self, params):
+    def import_fasta_mass(self, params, threads_per_cpu, max_threads, max_cumsize=None, parallelize=True):
```

Comment:
This is a nice to have, so up to you, but I'd move the default constants for threads/cpu and max_threads to this class, import them in the impl file, and use them to set defaults here rather than making them required arguments.

Reply:
I thought we wanted to do catalog type check in the impl file and input validity check in the fta file.

What makes you change your mind? because you are concerned about the extra required params(threads_per_cpu, max_threads, max_cumsize) passed into import_fasta_mass?

How do you want self.threads/cpu and self.max_threads from impl to pass in if not making them required arguments?
hide them in params?

Reply:
The checks shouldn't change at all. All I'm saying is use the same defaults you're already using in the impl for default arguments in FtA like:

```suggestion
    def import_fasta_mass(self, params, threads_per_cpu=THREADS_PER_CPU, max_threads=MAX_THREADS, max_cumsize=None, parallelize=True):
```

Reply:
Updated!

Reply:
:+1: